### PR TITLE
Support complex constants and functions in fuse

### PR DESCRIPTION
**Pull request by kmaehashi**
Feature requested in #1072.

### CODE REVIEW

**Comment by asi1024** [comment]
Jenkins test this please.

File: cupy/core/fusion.py [hunk 1]
```diff
@@ -392,11 +388,16 @@ def _get_out_params(var_list):
...
+    elif isinstance(var.const, complex):
+        init = '(%s, %s)' % (var.const.real, var.const.imag)
+    elif isinstance(var.const, (int, float)):
+        init = '= %s' % str(var.const)
     else:
-        return 'const %s v%d = %s;\n' % (
-            _dtype_to_ctype[var.ty],
-            var.num,
-            _const_to_str(var.const))
+        raise TypeError('Invalid constant type: {}'.format(type(var.const)))
```
Comment (asi1024) [inline cupy/core/fusion.py]: An error occurs when `var.const` is NumPy scalar value.

**Comment by kmaehashi** [comment]
I've merged your fix in https://github.com/kmaehashi/cupy/pull/1. Thanks!

**Comment by kmaehashi** [comment]
Note: this PR also fixes a bug in cupy.fuse that numpy scalar of `bool_` type did not work.

File: cupy/core/fusion.py [hunk 2]
```diff
@@ -264,18 +264,14 @@ def copy(self):
...
-    return str(val).lower() if type(val) is bool else str(val)
+_dtype_list = [numpy.dtype(_) for _ in '?bhilqBHILQefdFD']
 
 
 def _normalize_arg(arg, mem):
     arg_type = type(arg)
     if arg_type is _FusionRef:
         return arg._var
-    is_scalar = arg_type in [int, float, bool]
+    is_scalar = arg_type in [int, float, bool, complex]
```
[the remark this answers is not in the document]
Comment (kmaehashi) [reply cupy/core/fusion.py]: Thanks, fixed.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**